File: mobile/ios/shotclock/ShotClock/Models/SessionPlayer.swift
import Foundation

struct SessionPlayer: Codable, Identifiable {
    let id: String
    let user: PlayerUser
    let joinedAt: String
    let isAdmin: Bool
}

struct PlayerUser: Codable {
    let id: Int
    let username: String
    let displayName: String?

    var preferredName: String {
        if let displayName, !displayName.isEmpty { return displayName }
        return username
    }
}

File: mobile/android/shotclock/app/src/main/java/fm/shotclock/mobile/model/SessionPlayer.kt
package fm.shotclock.mobile.model

import fm.shotclock.mobile.data.network.dto.SessionPlayerDto

data class SessionPlayer(
    val id: String,
    val userId: Int,
    val username: String,
    val displayName: String,
    val isAdmin: Boolean,
)

fun SessionPlayerDto.toDomain(): SessionPlayer = SessionPlayer(
    id = id,
    userId = user.id,
    username = user.username,
    displayName = user.displayName.orEmpty().ifBlank { user.username },
    isAdmin = isAdmin,
)
